mako/codegen.py

```diff
@@ -785,25 +788,48 @@ def locate_encode(name):
             else:
                 return filters.DEFAULT_ESCAPES.get(name, name)
 
-        if "n" not in args:
+        filter_args = set()
+        if DEFAULT_ESCAPED_N not in args:
             if is_expression:
                 if self.compiler.pagetag:
                     args = self.compiler.pagetag.filter_args.args + args
-                if self.compiler.default_filters and "n" not in args:
+                    filter_args = set(self.compiler.pagetag.filter_args.args)
+                if (
+                    self.compiler.default_filters
+                    and DEFAULT_ESCAPED_N not in args
+                ):
                     args = self.compiler.default_filters + args
         for e in args:
-            # if filter given as a function, get just the identifier portion
-            if e == "n":
+            if e in (DEFAULT_ESCAPED_N, "n"):
```

Comment:
The changes here fix the problem mentioned in [issue 100](https://github.com/sqlalchemy/mako/issues/415).